docs/pages/providers/battlenet.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Battle.net"
+---
+
+# Battle.net
+
+OAuth 2.0 provider for Battle.net.
+
+Also see the [OAuth 2.0](/guides/oauth2) guide.
+
+## Initialization
+
+```ts
+import { BattleNet } from "arctic";
+
+const battlenet = new BattleNet(clientId, clientSecret, redirectURI);
+```
+
+## Create authorization URL
+
+```ts
+import { generateState } from "arctic";
+
+const state = generateState();
+const scopes = ["openid", " wow.profile "];
+const url = battlenet.createAuthorizationURL(state, scopes);
+```
+
+## Validate authorization code
+
+`validateAuthorizationCode()` will either return an [`OAuth2Tokens`](/reference/main/OAuth2Tokens), or throw one of [`OAuth2RequestError`](/reference/main/OAuth2RequestError), [`ArcticFetchError`](/reference/main/ArcticFetchError), or a standard `Error` (parse errors).
```

Comment:
With v3, I've added a [UnexpectedResponseError](https://arcticjs.dev/reference/main/UnexpectedResponseError) and  [UnexpectedErrorResponseBodyError](https://arcticjs.dev/reference/main/UnexpectedErrorResponseBodyError). See the other provider's docs